src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java

```diff
@@ -89,40 +91,80 @@ public void update()
                     {
                         checkInWareHouse((AbstractBuildingWorker) buildingEntry.getValue(), true);
                     }
+                    else if(buildingEntry.getValue() instanceof BuildingHome && ((BuildingHome) buildingEntry.getValue()).isFoodNeeded())
+                    {
+                        checkInWareHouseForFood((BuildingHome) buildingEntry.getValue(), true);
+                    }
                     this.index++;
                 }
                 i++;
             }
         }
     }
 
+    public boolean checkInWareHouseForFood(final BuildingHome buildingEntry, final boolean addToList)
+    {
+        if (buildingEntry.isFoodNeeded())
+        {
+            if (isInHut(itemStack -> !InventoryUtils.isItemStackEmpty(itemStack) && itemStack.getItem() instanceof ItemFood))
+            {
+                if (addToList)
+                {
+                    buildingEntry.setOnGoingDelivery(true);
+                    list.add(buildingEntry);
+                }
+                return true;
+            }
+
+            if (list.contains(buildingEntry))
+            {
+                list.remove(buildingEntry);
+                buildingEntry.setOnGoingDelivery(false);
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Get the first task in the list.
+     * @return the building which needs a delivery.
+     */
+    @Nullable
+    public AbstractBuilding getTask()
+    {
+        if(list.isEmpty())
+        {
+            return null;
+        }
+        return list.remove(0);
+    }
+
     /**
      * Check if the required items by the building are in the wareHouse.
-     *
      * @param buildingEntry the building requesting.
-     * @param addToList     if is in warehouse should add to the list?
+     * @param addToList if is in warehouse should add to the list?
      * @return true if has something in warehouse to deliver.
      */
     public boolean checkInWareHouse(@NotNull final AbstractBuilding buildingEntry, boolean addToList)
     {
-        if (buildingEntry.areItemsNeeded())
+        if(buildingEntry.areItemsNeeded())
         {
-            for (final ItemStack stack : buildingEntry.getNeededItems())
+            for(final ItemStack stack : buildingEntry.getNeededItems())
             {
-                if (stack == null || stack == ItemStack.EMPTY || (deliveryManHasBuildingAsTask(buildingEntry) && addToList))
+                if(stack == null || (deliveryManHasBuildingAsTask(buildingEntry) && addToList))
                 {
                     continue;
                 }
 
-                if (isInHut(stack))
+                if(isInHut(stack))
                 {
-                    if (addToList)
+                    if(addToList)
                     {
                         buildingEntry.setOnGoingDelivery(true);
                         list.add(buildingEntry);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "add()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)